From f7874e87d4ca9eafac452fa89a672da3b23bd874 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:20:24 -0400
Subject: [alsa] hda/ca0132: Fix recording from mode id 0x8

Message-id: <140972823573-129-patchset-jkysela@redhat.com>
Patchwork-id: 90829
O-Subject: [RHEL 7.1 PATCH 128/442] ALSA: hda/ca0132 - Fix recording from mode id 0x8
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Incorrect ADC is picked in ca0132_capture_pcm_prepare(),
where it assumes multiple streams while there is one stream
per ADC. Note that ca0132_capture_pcm_cleanup() already does
the right thing.

The Chromebook Pixel has a microphone under the keyboard that
is attached to node id 0x8. Before this fix, recording would
always go to the main internal mic (node id 0x7).

Signed-off-by: Hsin-Yu Chao <hychao@chromium.org>
Reviewed-by: Dylan Reid <dgreid@chromium.org>
Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 13c12dbe3a2ce17227f7ddef652b6a53c78fa51f)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_ca0132.c b/sound/pci/hda/patch_ca0132.c
index b9f2cbb..c7ea40f 100644
--- a/sound/pci/hda/patch_ca0132.c
+++ b/sound/pci/hda/patch_ca0132.c
@@ -2767,9 +2767,7 @@ static int ca0132_capture_pcm_prepare(struct hda_pcm_stream *hinfo,
      unsigned int format,
      struct snd_pcm_substream *substream)
 {
- struct ca0132_spec *spec = codec->spec;
-
- snd_hda_codec_setup_stream(codec, spec->adcs[substream->number],
+ snd_hda_codec_setup_stream(codec, hinfo->nid,
        stream_tag, 0, format);
 
  return 0;
-- 
1.7.1